src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -114,6 +114,11 @@ public function onKernelResponse(FilterResponseEvent $event)
             return;
         }
 
+        if (null !== $this->dispatcher) {
```

Comment:
Maybe: 

     if (null != $this->dispatcher && true === $this->registered) {

to be on the safe side?